### ISSUE TITLE
Very long time for send/receive mail via WiFi, no problem with mobile network

### ISSUE THREAD

**Issue by dirkhohmann** [thunderbird/thunderbird-android#5866]
### Checklist

- [X] I have used the search function to see if someone else has already submitted the same bug report.
- [X] I will describe the problem with as much detail as possible.

### App version

5.806

### Where did you get the app from?

Google Play

### Android version

11 RKQ1.200826.002

### Device model

Xiaomi Poco X3 Pro

### Steps to reproduce

pull down screen to receive mails or attempt to send mails

### Expected behavior

Action executed within seconds

### Actual behavior

Very long time or failure on WiFi. Normal behaviour on mobile network.
I tested same mail with Notebook connected to same WiFi - usual behaviour.
Router is only a few meters away.

### Logs

[k9_log.txt](https://github.com/k9mail/k-9/files/7875105/k9_log.txt)

**Comment by cketti** [comment]
The log doesn't contain any log messages from K-9 Mail. Did you check `Settings → General settings → Debugging → Enable debug logging` before checking for new messages?

**Comment by dirkhohmann** [comment]
[k9.log](https://github.com/k9mail/k-9/files/7875520/k9.log)
Sorry, picked wrong file.

**Comment by dirkhohmann** [comment]
By the way. I have the same issue on my old Huawei Phone.
It worked well for a long time. Maybe a recent fix causing that problem.

**Comment by cketti** [comment]
The issue is that the name of your incoming mail server resolves to both an IPv6 and an IPv4 address. But your mail service is only listening on IPv4, not IPv6. Currently K-9 Mail tries to sequentially connect to the addresses in the order they are returned from the OS (most are configured to prefer IPv6).

In your case K-9 Mail tries to connect to to the IPv6 address first. That connection attempt times out after 30 seconds. Then the app successfully connects to the IPv4 address. Unfortunately, that means it takes at least 30 seconds for a connection to be established.

The proper fix is for us to implement the "[happy eyeballs](https://en.wikipedia.org/wiki/Happy_Eyeballs)" algorithm to connect to both IP addresses in parallel and use the connection that can be established first.

What you could do: 
- Configure the mail server to also listen on IPv6. If you're not hosting the server yourself, you'd have to ask your email provider for this. I don't think it's an unreasonable request. And it's something that needs to happen sooner or later anyway.
- Expert/annoying option: use a host name that only resolves to the IPv4 address of your mail server. This will lead to a certificate error when changing settings in K-9 Mail, because the host name configured in the app doesn't match the hostname in the server's certificate. K-9 Mail allow users to manually accept those certificates. However, you'll have to repeat this step each time the server renews its certificate.

Duplicate of #1382

**Comment by cketti** [comment]
Correction: The configured mail server does listen on IPv6. It's probably an issue with your network, then.

**Comment by dirkhohmann** [comment]
Thanks for the fast reply.
But I didn't get it completely.
What ipv is missing 4 or 6 ? 
Probably the stupid WLAN router doesn't handle it right.
The server belongs to me, of course I could also change the DNS-records,but everything else seems to work fine.

**Comment by dirkhohmann** [comment]
Thanks for the fast reply.
Yes I see the probleminthe log.
Must find a way to fix it :-(
